providers/flagd-in-process/pkg/provider.go

```diff
@@ -646,3 +624,16 @@ func syncProviderFromConfig(logger *logger.Logger, sourceConfig runtime.SourceCo
 			sourceConfig.Provider, SourceTypeGrpc, SourceTypeKubernetes)
 	}
 }
+
+func mapError(err error) of.ResolutionError {
```

Comment:
@bacherfl @Kavindu-Dodan this is the only functional change in this PR - as the description says, the errors thrown in the json-evaluator are just errors, so they need to be handled this way (flagd itself has similar code for mapping them to connect errors).

Reply:
Yup, this fully matches with flagd error handling - https://github.com/open-feature/flagd/blob/main/core/pkg/service/flag-evaluation/flag_evaluator.go#L316